homeassistant/components/lcn/climate.py

```diff
@@ -107,9 +108,10 @@ def supported_features(self) -> int:
         return const.SUPPORT_TARGET_TEMPERATURE
 
     @property
-    def temperature_unit(self) -> str:
+    def temperature_unit(self) -> UnitTemperatureT:
         """Return the unit of measurement."""
-        return cast(str, self.unit.value)
+        # Config schema only allows for: TEMP_CELSIUS and TEMP_FAHRENHEIT
+        return cast(UnitTemperatureT, self.unit.value)
```

Comment:
This isn't really correct. The unit values are defined in the library and could be changed to not match our expected definition. Preferably the integration should be updated to have some logic to return one of the Home Assistant temperature constants instead.

Reply:
Good point. I've updated the code to return either `TEMP_FAHRENHEIT` or `TEMP_CELSIUS` based on `self.unit`.

For reference the library code: [pypck/lcn_defs.py -> VarUnit(Enum)](https://github.com/alengwenus/pypck/blob/e68cf05d968756adc6aea7b8a1629c6c67a8a8ed/pypck/lcn_defs.py#L675-L677)